deepchem/feat/molecule_featurizers/molgan_featurizer.py

```diff
@@ -0,0 +1,214 @@
+import logging
+import numpy as np
+import rdkit.Chem as Chem
+
+from deepchem.utils.typing import RDKitBond, RDKitMol, List
+from deepchem.feat.base_classes import MolecularFeaturizer
+
+logger = logging.getLogger(__name__)
+
+
+class GraphMatrix:
+    """
+    This is class used to store data for MolGAN neural networks.
+
+    Parameters
+    ----------
+    node_features: np.ndarray
+      Node feature matrix with shape [num_nodes, num_node_features]
+    edge_features: np.ndarray,
+      Edge feature matrix with shape [num_nodes, num_nodes]
+
+    Returns
+    -------
+    graph: GraphMatrix
+      A molecule graph with some features.
+    """
+
+    def __init__(self, adjacency_matrix: np.ndarray, node_features: np.ndarray):
+        self.adjacency_matrix = adjacency_matrix
+        self.node_features = node_features
+
+
+class MolGanFeaturizer(MolecularFeaturizer):
+    """This class implements featurizer used with MolGAN de-novo molecular generation based on:
+    `MolGAN: An implicit generative model for small molecular graphs`<https://arxiv.org/abs/1805.11973>`_.
```

Comment:
Could you add the link to a `Reference` section in the docstring as shown in https://numpydoc.readthedocs.io/en/latest/format.html? (See section 14.)